### PR TITLE
Use new eval and function info endpoints

### PR DESCRIPTION
**Pull request by stardust66**
Conform to the suggestions made in https://github.com/concord-consortium/codap/pull/390.

### CODE REVIEW

File: src/utils/codapPhone/index.ts [hunk 1]
```diff
@@ -885,12 +886,14 @@ export const getFunctionNames: () => Promise<string[]> = (() => {
...
           action: CodapActions.Get,
-          resource: CodapResource.FunctionNames,
+          resource: CodapResource.FormulaEngine,
         },
-        (response: GetFunctionNamesResponse) => {
+        (response: GetFunctionInfoResponse) => {
           if (response.success) {
-            names = response.values;
-            resolve(response.values);
+            const allFunctions: typeof response.values["category"]["fname"][] =
```
Comment (P-bibs) [inline src/utils/codapPhone/index.ts:893]: Would it be worth splitting off the inner type of the response from `FormulaEngine` to simplify the type expression here?

**Review by P-bibs — APPROVED** [review]
Looks good, just one comment.

**Review by thomascastleman — APPROVED** [review]

File: src/utils/codapPhone/index.ts (hunk 1 above)
Comment (stardust66) [reply src/utils/codapPhone/index.ts]: Agreed. This should be better.

**Comment by P-bibs** [comment]
Looks good :+1:

**Pull request merged**